lib/signers/presign.js

```diff
@@ -49,31 +53,31 @@ function signedUrlSigner(request) {
     queryParams = AWS.util.queryStringParse(parsedUrl.search.substr(1));
   }
 
-  AWS.util.each(request.httpRequest.headers, function (key, value) {
-    if (key === expiresHeader) key = 'Expires';
-    if (key.indexOf('x-amz-meta-') === 0) {
-      // Delete existing, potentially not normalized key
-      delete queryParams[key];
-      key = key.toLowerCase();
-    }
-    queryParams[key] = value;
-  });
-  delete request.httpRequest.headers[expiresHeader];
-
-  var auth = queryParams['Authorization'].split(' ');
+  var auth = request.httpRequest.headers['Authorization'].split(' ');
   if (auth[0] === 'AWS') {
     auth = auth[1].split(':');
     queryParams['AWSAccessKeyId'] = auth[0];
     queryParams['Signature'] = auth[1];
+
+    AWS.util.each(request.httpRequest.headers, function (key, value) {
+      if (key === expiresHeader) key = 'Expires';
+      if (key.indexOf('x-amz-meta-') === 0) {
```

Comment:
Do you know if there are any headers that we hoist to the query string today that don't also have to be sent as headers when using sigv4 presigned urls? Maybe that's just a feature of sigv2, but I'm wondering if we'll lose out on that feature with this change.

Reply:
There are, and they are all handled [here](https://github.com/aws/aws-sdk-js/blob/master/lib/signers/v4.js#L44). The current V4 presigner will always produce an invalid signature when any other headers are present.

Reply:
Ah ok. So we may want to add additional headers to that section you pointed to in the future, but as of now, this change won't cause any existing cases to break. This part looks good then!